operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -276,10 +280,41 @@ func createManifestWorks(c client.Client, restMapper meta.RESTMapper,
 					container.Env[j].Value = strconv.FormatBool(installProm)
 				}
 			}
+
+			if hasCustomRegistry {
+				oldImage := container.Image
+				newImage, err := imageRegistryClient.Cluster(clusterName).ImageOverride(oldImage)
+				if err == nil {
+					spec.Containers[i].Image = newImage
+				}
+			}
 		}
 	}
+
 	manifests = injectIntoWork(manifests, dep)
 
+	// replace the pull secret and addon components image
+	if hasCustomRegistry {
+		customPullSecret, err := imageRegistryClient.Cluster(clusterName).PullSecret()
+		if err == nil && customPullSecret != nil {
+			pullSecret = customPullSecret
+		}
+
+		images := imageListConfigMap.Data
+		for key, oldImage := range images {
+			newImage, err := imageRegistryClient.Cluster(clusterName).ImageOverride(oldImage)
+			if err == nil {
+				imageListConfigMap.Data[key] = newImage
+			}
+		}
+	}
+
+	if pullSecret != nil {
```

Comment:
is it possible the custom pull sectet has conflict with the default  pull secret?

Reply:
a little confused by the logic for the custom pull secret.
the customPullSecret will use the same name as the original pullSecret, and replace the original one? Or co-exist with different name.
Also, in current logic, pullSecret is a global var, if it's updated in one manifestwork, seems it will affect the following ones?

Reply:
@marcolan018 thanks for your suggestions, let me check it.